File: Hash Table/1-Two-Sum/1-Two-Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> comp = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length ; i++)
        {
            if (comp.containsKey(target-nums[i]))
            {
				return new int[] {i, comp.get(target-nums[i])};
            }
            else
            {
                comp.put(nums[i], i);
            }
        }
        return null;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;
        vector<int> v;
        for(int i=0; i< nums.size(); i++){
            if(um.find(target-nums[i])!= um.end()){
                v.push_back(i);
                v.push_back(um[target-nums[i]]);
                break;
            }
            
            
            um[nums[i]]= i;
            
        }
        return v;
    }
};